src/Symfony/Component/Serializer/Normalizer/AbstractNormalizer.php

```diff
@@ -302,21 +302,8 @@ protected function getConstructor(array &$data, $class, array &$context, \Reflec
      *
      * @throws RuntimeException
      */
-    protected function instantiateObject(array &$data, $class, array &$context, \ReflectionClass $reflectionClass, $allowedAttributes/*, $format = null*/)
+    protected function instantiateObject(array &$data, $class, array &$context, \ReflectionClass $reflectionClass, $allowedAttributes, ?string $format = null)
```

Comment:
Nico, is true that these two look identical:

```php
public function foo(string $foo = null);
public function foo(?string $foo = null);
```

but I agree with Maxime that they are different because only the second one lets you change the default value without breaking the apps:

```php
// will break for apps who previously used null for the foo argument
public function foo(string $foo ='bar');

// will keep working regardless of the value of the foo argument
public function foo(?string $foo = 'bar');
```